gb-cpu/src/address/mod.rs

```diff
@@ -0,0 +1,20 @@
+pub mod area;
+pub mod consts;
+
+use area::Area;
+use consts::*;
+
+pub fn relative(area: Area, address: u16) -> usize {
```

Comment:
That's the point that I want to talk about when i'm talking about memory usage, I think that using a function is better that instanciating a struct in this case. 

What I think is better isn't necesserly the case, but I want to suggest it and have a discussion about the pros and cons of each way

Reply:
my point is when I implement a controller (i.e.: romonly, mbc1) 
I always use the relative address and never the absolute one (because I think a controller shouldn't relie one the absolute address given by the `memorybus` nor using absolute offset inside the code of the controller)
it'll add a lot of redundant call to `relative()` with your choice.

Reply:
I don't think you quite see the flow of the data in my proposition: there's only one place where you need to translate from absolute to relative, in the memory.read [See Here](https://github.com/Ragarnoy/GBMU/blob/ea6eee5ed35ff9c804de534c61124c7ae194f295/gb-cpu/src/memory/mod.rs#L31). Never in the mbcs or the different memory spaces will you have the absolute adress.

As you said, your mbc doesn't care at all about the absolute one. Thats why rather than storing both value at all time, i suggest to just do the translation at that point, and each module do it's own work in it's relative space.

Reply:
I see, I think it could be better to have this function defined as a method of `Area` ?

Reply:
> I don't think you quite see the flow of the data in my proposition: there's only one place where you need to translate from absolute to relative, in the memory.read  

I disagree on that point. I think the translation from absolute to relative should be done once by the AddressBus before calling `FileOperation.read`, otherwise if the translation is done in the read function every implementation of FileOperation will have to do it.

> Thats why rather than storing both value at all time, i suggest to just do the translation at that point, and each module do it's own work in it's relative space.  

Storing both the absolute and relative in the `Adress` can be useful in case errors. If a `FileOperation` need to return an error, having the absolute value will be more useful than only the relative adress.

Reply:
Great suggestion, it makes more sense as a function method of Area!

@guilhemSmith the memory.read is eq to the adress bus, so we agree on that

You don't need to store it a second time: as the error handling is done at each stack frame, you still have access to the absolute address when you return the error at the memory.read level